pxtblocks/code-validation/rubricCriteria.ts

```diff
@@ -0,0 +1,56 @@
+namespace pxt.blocks {
+    interface BlockSet {
+        blocks: string[];
+        count: number;
+    }
+
+    export interface CriteriaData {
+        criteriaID: string;
+        friendlyText: string;
+        blockRequirements?: BlockSet[];
+        count?: number;
+    }
+
+    export function getCriteria(data: CriteriaData): RubricCriteria {
+        switch (data.criteriaID) {
+            case "blockCheck":
+                return new BlockCheckCriteria(data.friendlyText, data.blockRequirements);
+            case "comment":
+                return new CommentCriteria(data.friendlyText, data.count);
+            default:
+                console.error(`Unrecognized criteriaID: ${data.criteriaID}`);
+                return null;
+        }
+    }
+
+    export abstract class RubricCriteria {
+        friendlyText: string;
```

Comment:
Re `friendlyText`: How will localization work for rubric content? Now I'm curious what we do for tutorials (I don't know off hand). We might want to follow that pattern.

Reply:
I have no idea...

Reply:
Vaguely, I think the markdown files for tutorials are checked in, then those are parsed and somehow sent for translation? But if these aren't going to be checked into our repo, that may not work.

For teacher-generated rubrics, I don't think translation should be a goal.

For rubrics we provide, I'm wondering if we could just provide different files/links for the different languages, then teachers select the ones that are most appropriate for their students? It depends on how we end up storing the rubrics, I suppose. I wonder if our existing curriculum content would be a better pattern to follow, rather than tutorials. Seems like that might be a slightly more similar scenario?